pkg/cli/completion.go

```diff
@@ -85,6 +86,98 @@ func (a *completion) complete(cmd *cobra.Command, args []string, toComplete stri
 	return removeExistingArgs(result, args), a.successDirective
 }
 
+// checkProjectPrefix
+// called on a completion object, which already has the completion function defined
+// this function modifies that completion function, wraps it with some logic to handle projects
+func (a *completion) checkProjectPrefix() *completion {
+	currentCompletionFunc := a.completionFunc
+	foundCompletions := []string{}
+	// Get CLI Config, necessary for building new clients on different projects
+	options := a.client.Options()
+	if options.CLIConfig == nil {
+		cfg, err := config.ReadCLIConfig()
+		options.CLIConfig = cfg
+		if err != nil {
+			fmt.Printf("could not read CLI config.\n%+v", err)
+		}
+	}
+	newCompletionFunc := func(ctx context.Context, c client.Client, toComplete string) ([]string, error) {
+		if parsedProject, after, found := strings.Cut(toComplete, "::"); found {
+			// check that the provided project exists, return a helpful error otherwise
+			foundProject, err := c.ProjectGet(ctx, parsedProject)
+			if foundProject == nil || err != nil {
+				return []string{}, nil
```

Comment:
We should be returning the error in this case, right?
```suggestion
				return []string{}, err
```

Reply:
Hm, curious what we'd actually want in this case. 

Say a user is autocompleting DoesNotExist::ok, should this throw an error, or just fail to autocomplete?

It currently doesn't return an error, to have the function fail to autocomplete without an error.

Reply:
Ah, thanks for clearing this up for me. The way it currently is written makes more sense. Thank you!